realm/realm-library/src/main/java/io/realm/BaseRealm.java

```diff
@@ -162,6 +162,9 @@ public void removeChangeListener(RealmChangeListener<? extends BaseRealm> listen
      * }
      * </pre>
      *
+     * <p>Note that when the Realm is accessed from threads other than where it was created, {@link IllegalStateException}
```

Comment:
This issue is about RealmObject not Realm, so this should ideally be moved to RealmResults/RealmObject instead.